front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/SpeedSpaceChart.tsx

```diff
@@ -33,21 +51,17 @@ const CHART_MIN_HEIGHT = 250;
  * - 2 marchs displayed: base and alternative
  *
  */
-export default function SpeedSpaceChart(props) {
-  const {
-    dispatchUpdateMustRedraw,
-    dispatchUpdateTimePositionValues,
-    initialHeight,
-    positionValues,
-    trainSimulation,
-    simulationIsPlaying,
-    speedSpaceSettings,
-    onSetSettings,
-    onSetChartBaseHeight,
-    timePosition,
-  } = props;
+export default function SpeedSpaceChart({
+  initialHeight,
+  onSetChartBaseHeight,
+}: SpeedSpaceChartProps) {
+  const selectedTrain = useSelector(getSelectedTrain);
+  const timePosition = useSelector(getTimePosition);
+  const positionValues = useSelector(getPositionValues);
```

Comment:
Would it be possible to replace these 3 selectors by props ? The parent component already has selectedTrain, so you would just have to move the selector of timePosition and positionValues to the parent component. This way, you can be sure selectedTrain is not undefined (and maybe we can rename it train simply ?)

-> you can use noop in the isolatedEnableInteractivity

Reply:
Actually I started to put all the selectors in parent (SimulationResult) to match the new front architetcure but if I do that, it means OSRDSdtcmResult, who also calls SpeedSpaceChart, needs to have these selectors. But as OSRDSdtcmResult is not a view (the view is 2 layers above it), it would have meant to pass a lot of props and at that time, I wasn't sure it was ok to pass down props on 4 or 5 layers.

Reply:
As discussed, I removed selectedTrain from SpeedSpace to use this selector in SimulationResult.
I added the selector in OSRDStdcmResults.
But I left the conversion to trainSimulation in SpeedSpace as it's used only in this component.